Several examples do not compile.

When I build entire G+Smo with gsUnstructuredSplines with `make` (using gcc 11.2), there is an example from gsUnstructuredSplines that does not compile and several warnings.

- [x] kirchhoff-Love_weak_multipatch_buckling_example.cpp
```
In file included from [...]/gismo/optional/gsUnstructuredSplines/examples/kirchhoff-Love_weak_multipatch_buckling_example.cpp:23:0:
[...]/gismo/optional/gsSpectra/gsSpectra.h:23:10: fatal error: Spectra/include/Spectra/SymEigsSolver.h: No such file or directory
 #include <Spectra/include/Spectra/SymEigsSolver.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

- [x] biharmonic2_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic2_eigenvalue_example.cpp:243:13: warning: unused variable ‘systemRet’ [-Wunused-variable]
         int systemRet = system("mkdir -p ModalResults");
             ^~~~~~~~~
```
- [x] biharmonic2_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic2_eigenvalue_example.cpp:273:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     GISMO_ENSURE((omegas.size()>=values.rows()),"Too few analytical eigenvalues");
                   ~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```
- [x] biharmonic2_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic2_eigenvalue_example.cpp:281:13: warning: unused variable ‘systemRet’ [-Wunused-variable]
         int systemRet = system("mkdir -p ModalResults");
             ^~~~~~~~~
```
- [x] create_multipatch.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/create_multipatch.cpp:110:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (index_t p=0; p!=mp.nPatches(); p++)
                               ~^~~~~~~~~~~~~~~
```
- [x] create_multipatch.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/create_multipatch.cpp:112:28: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
                 if (hbasis = dynamic_cast<gsHTensorBasis<2,real_t> *>(&mp.basis(p)))
                     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
- [x] create_multipatch.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/create_multipatch.cpp:127:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (index_t p=0; p!=mp.nPatches(); p++)
                                   ~^~~~~~~~~~~~~~~
```
- [x] biharmonic_planar_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_example.cpp:416:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (index_t p = 0; p != geom.nPatches(); p++)
                                     ~~^~~~~~~~~~~~~~~~~~
```
- [x] biharmonic_planar_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_example.cpp:487:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for (index_t p = 0; p != geom0.nPatches(); p++)
                                     ~~^~~~~~~~~~~~~~~~~~~
```
- [x] biharmonic_planar_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:553:34: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for (index_t p = 0; p!=mp.nPatches(); p++)
                                 ~^~~~~~~~~~~~~~~
```
- [x] biharmonic_planar_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:539:13: warning: unused variable ‘systemRet’ [-Wunused-variable]
         int systemRet = system("mkdir -p ModalResults");
             ^~~~~~~~~
```
- [x] biharmonic_planar_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:572:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     GISMO_ENSURE(omegas.size()>=values.rows(),"Too few analytical eigenvalues");
                  ~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```
- [x] biharmonic_surface_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:580:13: warning: unused variable ‘systemRet’ [-Wunused-variable]
         int systemRet = system("mkdir -p ModalResults");
             ^~~~~~~~~
```
- [x] biharmonic_planar_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:178:10: warning: unused variable ‘last’ [-Wunused-variable]
     bool last = false;
          ^~~~
```
- [x] biharmonic_planar_eigenvalue_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_planar_eigenvalue_example.cpp:180:10: warning: unused variable ‘cond’ [-Wunused-variable]
     bool cond = false;
          ^~~~
```
- [x] biharmonic_surface2_example.cpp
```
[...]/gismo/optional/gsUnstructuredSplines/examples/biharmonic_surface2_example.cpp:1069:10: warning: variable ‘laplace_u_ex’ set but not used [-Wunused-but-set-variable]
     auto laplace_u_ex = ev.getVariable(laplace, G);
          ^~~~~~~~~~~~
```

--- a/examples/kirchhoff-Love_weak_multipatch_buckling_example.cpp
+++ b/examples/kirchhoff-Love_weak_multipatch_buckling_example.cpp
@@ -20,7 +20,9 @@
 #include <gsKLShell/src/gsThinShellUtils.h>
 #endif
 
+#ifdef gsSpectra_ENABLED
 #include <gsSpectra/gsSpectra.h>
+#endif
 
 using namespace gismo;
 

